neasqc_wp61/data/data_processing/use_alpha_3_4.py

```diff
@@ -230,167 +221,204 @@ def main(args) -> dict:
+    param_init_functions = [
+        ("rescaled_unif", torch.nn.init.uniform_),  # (a=0, b=2 * np.pi)
+        ("unif", torch.nn.init.uniform_),  # (a=0, b=2 * np.pi)
+        ("norm", torch.nn.init.normal_),  # (mean=np.pi, std=np.sqrt(np.pi))
+        ("none", None),
+    ]
 
+    error = True
     for param_init_function in param_init_functions:
-        if args.circuit_init[0] == param_init_function[0]:
-            circuit_init = param_init_functions[1]
-        else:
-            raise ValueError("Unrecognised qubit init function.")
-        
+        if args.circuit_init == param_init_function[0]:
+            circuit_init = param_init_function[1]
+            error = False
+            break
+    if error:
+        raise ValueError("Unrecognised qubit init function.")
+
+    error = True
     for param_init_function in param_init_functions:
-        if args.data_rescaling[0] == param_init_function[0]:
-            data_rescaling = param_init_functions[1]
-        else:
-            raise ValueError("Unrecognised data rescaling function.")
-        
+        if args.data_rescaling == param_init_function[0]:
+            data_rescaling = param_init_function[1]
+            error = False
+            break
+    if error:
+        raise ValueError("Unrecognised data rescaling function.")
+
+    error = True
     for param_init_function in param_init_functions:
-        if args.mlp_init[0] == param_init_function[0]:
-            mlp_init = param_init_functions[1]
-        else:
-            raise ValueError("Unrecognised MLP init function.")
+        if args.mlp_init == param_init_function[0]:
+            mlp_init = param_init_function[1]
+            error = False
+            break
+    if error:
+        raise ValueError("Unrecognised MLP init function.")
 
     # Prepare data
 
+    # Set dataset
+    if args.dataset == "ag_news":
+        n_classes = 4
+        pass
+    elif args.dataset == "newscatcher":
+        n_classes = 7
+        pass
+    elif args.dataset == "reviews":
+        n_classes = 3
+        pass
+    elif args.dataset == "bert_test":
+        n_classes = 2
+        dataset_path = "../datasets/bert_test_pipeline_dataset.pkl"
+
     sentence_vectors = []
     labels = []
-    dataset_path = args.dataset
     split_id = args.split_id
 
-    split_data_path = dataset_path.split('_')
-    split_data_path = [item.split('.') for item in split_data_path]
-
-    if args.embedder not in split_data_path:
-            raise ValueError("The dataset provided does not correspond"
-                             " to the choice of embedder.")
+    if args.embedder.lower() not in dataset_path:
+        raise ValueError(
+            "The dataset provided does not correspond"
+            " to the choice of embedder."
+        )
 
     dataset = pd.read_pickle(dataset_path)
     dim_red.fit(data_to_fit=dataset)
```

Comment:
In the end Bruno joined only train + val. Test will be separate, i.e. the test file given by Tilde in their repo. This is to prevent the model from ever seeing any test data. Given that testing will be done by a separate script and on a separate dataset, I removed it. Now the full dataset is train+val with 10-fold cross-validation